### PR TITLE
Danfernand/init

### PR DESCRIPTION
**Pull request by danfernand**
adding better examples page, making sure travis CI works correctly and minor refactoring

### CODE REVIEW

File: examples/components/field-wrappers/material/input-wrapper.js [hunk 1]
```diff
@@ -40,10 +38,12 @@ class InputWrapper extends Component {
       onFocus: this.onFocus.bind(this),
       onChange: this.onChange.bind(this),
       value: this.props.value,
+      multiLine: this.props.multiLine,
+      rows: this.props.rows,
       floatingLabelText: this.props.floatingLabelText,
       hintText: this.props.hintText,
     };
-    if (this.state.shouldShowErrorMessage === true && errors.length > 0) {
+    if (this.state.shouldShowErrorMessage === true && errors.length > 0 && props.value !== undefined) {
```
Comment (dkoston) [inline examples/components/field-wrappers/material/input-wrapper.js]: this is getting pretty complex. I'd split out into a formHasErrors() function or similar to make sure others understand your intent

File: examples/constraints.js [hunk 2]
```diff
@@ -1,5 +1,38 @@
...
+
+function nameMessage(value, attribute) {
+  const nameAttribute = attribute === 'firstName' ? 'first name' : 'last name';
+  return validate.format('^%{name} is not a valid ' + nameAttribute + ', please enter a valid name', {
+    name: value
+  });
+}
+
+function passwordMessage() {
+  return 'password does not meet our guidelines, look below for more info';
```
Comment (dkoston) [inline examples/constraints.js]: this won't be below in context though. At runtime, you won't know what file caused this. Best to print a stacktrace or line number if you want people to view the source.

File: examples/containers/form1.js [hunk 3]
```diff
@@ -11,13 +11,21 @@ class Form1 extends FormComponent {
     this.formID = 'form1';
   }
   onFormFieldChanged() {
-    console.log(this.state.formData);
+    console.log(this.props.forms[this.formID].errors);
```
Comment (dkoston) [inline examples/containers/form1.js]: Please use debug instead of console.log so output can be toggled on/off: https://github.com/visionmedia/debug

Comment (danfernand) [inline examples/containers/form1.js]: done

File: examples/constraints.js (hunk 2 above)
Comment (danfernand) [inline examples/constraints.js]: forgot the password rules component that which what we are looking for in a password. added

File: examples/components/field-wrappers/material/input-wrapper.js (hunk 1 above)
Comment (danfernand) [inline examples/components/field-wrappers/material/input-wrapper.js]: done, logic in getErrorText

**Comment by danfernand** [comment]
Added dep to react-boostrap

**Comment by dkoston** [comment]
LGTM

**Pull request merged**